ynr/apps/sopn_parsing/helpers/extract_pages.py

```diff
@@ -18,6 +21,73 @@ def extract_pages_for_ballot(ballot, manual_upload=False):
     :type ballot: candidates.models.Ballot
 
     """
+    if ballot.sopn.uploaded_file.path.endswith(".pdf"):
+        pass
+    elif ballot.sopn.uploaded_file.path.endswith(".html"):
+        print(
+            f"Warning! HTML file types are not supported; trying to convert {ballot.sopn.uploaded_file.path} to pdf"
+        )
+
+        html_file_path = ballot.sopn.uploaded_file.path
+
+        with tempfile.NamedTemporaryFile(suffix=".pdf") as temp_file:
+            # convert the doc and save to a temp file
+            pypandoc.convert_file(
+                html_file_path, to="pdf", outputfile=temp_file.name
+            )
+
+            # save the pdf file to the sopn document
+            pdf_file_name = ballot.sopn.uploaded_file.name.split("/")[
+                -1
+            ].replace(".html", ".pdf")
+
+            ballot.sopn.uploaded_file.save(
+                name=pdf_file_name, content=temp_file, save=True
```

Comment:
Does using the `temp_file` directly for the content work? [The django docs](https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.FileField) suggested that it needs to be an in instance of its `File` object? So I would have thought you would need to do:
```
ballot.sopn.uploaded_file.save(
    name=pdf_file_name, content=File(temp_file), save=True
)
```

Reply:
It did work, but happy to make the change.

Reply:
UPDATE: tried saving it locally directly with the a tempfile and it worked, nice!